Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/bbox.h

```diff
@@ -255,6 +259,101 @@ namespace CGAL {
+    *       should be triangulated or not.
+    *       The default value is `false`, and faces are triangulated.}
+    *     \cgalParamDefault{false}
+    *   \cgalParamNEnd
+    *   \cgalParamNBegin{vertex_point_map}
+    *     \cgalParamDescription{a property map associating points to the vertices of `pmesh`}
+    *     \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+    *                    as key type and `%Point_3` as value type}
+    *     \cgalParamDefault{`boost::get(CGAL::vertex_point, pmesh)`}
+    *     \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+    *                     must be available in `PolygonMesh`.}
+    *   \cgalParamNEnd
+    *   \cgalParamNBegin{geom_traits}
+    *     \cgalParamDescription{an instance of a geometric traits class providing the functor `Construct_bbox_3`
+    *                           and the function `Construct_bbox_3 construct_bbox_3_object()`.
+    *                           `Construct_bbox_3` must provide `Bbox_3 operator()(Point_3)`
+    *                           where `%Point_3` is the value type of the vertex point map.
+    *                           It must also provide the functor `Construct_midpoint_3` and,
+    *                           the function `Construct_midpoint_3 construct_midpoint_3_object()`.
+    *                           `Construct_midpoint_3` must provide `Point_3 operator()(Point_3, Point_3)`.
+    *                           The types `Vector_3` and `Iso_cuboid_3` must also be provided.}
+    *   \cgalParamNEnd
+    * \cgalNamedParamsEnd
+    *
+    * @see `bbox()`
+    */
+    template<typename PolygonMesh,
+             typename NamedParameters = parameters::Default_named_parameters>
+    void add_bbox(PolygonMesh& pmesh,
+                  const NamedParameters& np = parameters::default_values())
+    {
+      using parameters::choose_parameter;
+      using parameters::get_parameter;
+
+      using GT = typename GetGeomTraits<PolygonMesh, NamedParameters>::type;
+      GT gt = choose_parameter<GT>(get_parameter(np, internal_np::geom_traits));
+      using Point_3 = typename GT::Point_3;
+      using Iso_cuboid_3 = typename GT::Iso_cuboid_3;
+      using Vector_3 = typename GT::Vector_3;
+
+      const double factor = choose_parameter(get_parameter(np, internal_np::bbox_scaling), 1.);
+      CGAL_precondition(factor > 0);
+      const bool dont_triangulate = choose_parameter(
+        get_parameter(np, internal_np::do_not_triangulate_faces), false);
+
+      const Iso_cuboid_3 bb(CGAL::Polygon_mesh_processing::bbox(pmesh, np));
+
+      auto midpoint = gt.construct_midpoint_3_object();
+      const Point_3 bb_center = midpoint((bb.min)(), (bb.max)());
+      const Iso_cuboid_3 bbext = (factor == 1.)
+        ? bb
+        : Iso_cuboid_3(bb_center + factor * Vector_3(bb_center, (bb.min)()),
```

Comment:
As you require `Construct_midpoint_3`  would it not be consequent to also require the functors for scaling and construction of the iso cuboid?